bikeapp/posts/routes.py

```diff
@@ -0,0 +1,61 @@
+
+from flask import (render_template, url_for, flash,
+                   redirect, request, abort, Blueprint)
+from flask_login import current_user, login_required
+from bikeapp import db
+from bikeapp.models import Post
+from bikeapp.posts.forms import PostForm
+
+posts = Blueprint('posts', __name__)
+
+
+@posts.route("/newPost", methods=['GET', 'POST'])
+@login_required
+def new_post():
+    form = PostForm()
+    if form.validate_on_submit():
+        post = Post(title=form.title.data, content=form.content.data, author=current_user)
+        db.session.add(post)
+        db.session.commit()
+        flash('Your post has been created!', 'success')
+        return redirect(url_for('main.home'))
+    return render_template('createPost.html', title='New Post',
+                           form=form, legend='New Post')
```

Comment:
Sometimes you can reduce nesting in the code using inverted-ifs.
```suggestion
    if not form.validate_on_submit():
       return render_template('createPost.html', title='New Post',
                           form=form, legend='New Post')
      post = Post(title=form.title.data, content=form.content.data, author=current_user)
      db.session.add(post)
      db.session.commit()
      flash('Your post has been created!', 'success')
      return redirect(url_for('main.home'))
```

Reply:
I actually don't have a problem with bigger if or else blocks. 
I prefer the condition is simpler to read rather than artificially negating it.
(be prepared that "both approaches" are valid here and you will most likely deal with people preferring one or the other... this is what consider "flavor remark")